config.md

```diff
@@ -384,31 +384,84 @@ For POSIX platforms, the configuration structure supports `hooks` for configurin
         * **`env`** (array of strings, OPTIONAL) with the same semantics as [IEEE Std 1003.1-2008's `environ`][ieee-1003.1-2008-xbd-c8.1].
         * **`timeout`** (int, OPTIONAL) is the number of seconds before aborting the hook.
             If set, `timeout` MUST be greater than zero.
+    * **`prestart-runtime`** (array of objects, OPTIONAL) is an array of [prestart-runtime hooks](#prestart-runtime-hooks).
+        Entries in the array have the same schema as pre-start entries.
+    * **`mount`** (array of objects, OPTIONAL) is an array of [mount hooks](#mount-hooks).
+        Entries in the array have the same schema as pre-start entries.
+    * **`start`** (array of objects, OPTIONAL) is an array of [start hooks](#start-hooks).
+        Entries in the array have the same schema as pre-start entries.
     * **`poststart`** (array of objects, OPTIONAL) is an array of [post-start hooks](#poststart).
         Entries in the array have the same schema as pre-start entries.
     * **`poststop`** (array of objects, OPTIONAL) is an array of [post-stop hooks](#poststop).
         Entries in the array have the same schema as pre-start entries.
 
 Hooks allow users to specify programs to run before or after various lifecycle events.
 Hooks MUST be called in the listed order.
-Hooks MUST be executed in the [runtime namespace](glossary.md#runtime-namespace).
 The [state](runtime.md#state) of the container MUST be passed to hooks over stdin so that they may do work appropriate to the current state of the container.
 
 ### <a name="configHooksPrestart" />Prestart
 
-The pre-start hooks MUST be called after the [`start`](runtime.md#start) operation is called but [before the user-specified program command is executed](runtime.md#lifecycle).
+The prestart hooks MUST be called after the [`start`](runtime.md#start) operation is called but [before the user-specified program command is executed](runtime.md#lifecycle).
+On Linux, for example, they are called after the container namespaces are created, so they provide an opportunity to customize the container (e.g. the network namespace could be specified in this hook).
+
+Note: Prestart hooks were deprecated in favor of prestart-runtime, mount and start hooks, which have a clearer definition.
+Additionally note that `runc` incorrectly implemented the prestart hooks, users should be warry of relying on a particular behavior (see [https://github.com/opencontainers/runc/issues/1710](https://github.com/opencontainers/runc/issues/1710) and [https://github.com/opencontainers/runc/pull/1811](https://github.com/opencontainers/runc/pull/1811)).
+
+Prestart hooks MUST be executed in the [runtime namespace](glossary.md#runtime-namespace).
```

Comment:
This should now be specified to refer either specifically to the deprecated `prestart` hook, or it is contradictory to the table below (where mount/start hooks are specified to be in the container namespace).

Reply:
Sorry I'm not sure I understand your comment :)

This statement should refer to the deprecated `prestart` hook because it is in the `prestart` section.

Reply:
I understood that only after re-reading, which is why I commented. I think the confusing part is that it refers to the plural "hooks" even though it talks about the single kind of `prestart` hook (of which there can be several), and *not* the several prestart hooks (as in collectively about the hooks available during the start phase) that are being introduced here.

I think one way to make this clearer would be to use the name as "`prestart` hooks" (and correspondingly for the other hooks).